### ISSUE TITLE
Add a warning when printing banners with `echo ###...`

### ISSUE THREAD

**Issue by J-M0** [koalaman/shellcheck#2747]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

echo #########################
echo Done!
echo #########################

```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

```
Line 3:
echo #########################
     ^-- SCXXX (warning): Are you trying to print a banner? Use quotes to actually display it.
```
---

I've been trying to implement this myself, but hit a block because `echo` and `echo ####` apparently parse to the same ast. I feel I could easily implement this with a regex, but I don't know if that's compatible with ShellCheck's architecture.

**Comment by aeiplatform** [comment]
This issue feels stupid.

Example: echo #Comment me
